Changes to make the gem a bit more modular

devonlambert opened the pull request:
The changes I've made allow for a bit more modularity of the generated code from the Gem. Wanted to share the changes and see what you thought.

wazery commented:
@devonlambert Thanks for the PR, will check it and let you know if I have any thoughts.

wazery commented on lib/generators/ratyrate/templates/jquery.raty.js.erb:
```diff
@@ -724,8 +724,8 @@
     cancelHint   : 'Cancel this rating!',
-    cancelOff    : '<%= asset_path('cancel-off.png') %>',
-    cancelOn     : '<%= asset_path('cancel-on.png') %>',
+    cancelOff    : 'cancel-off.png',
```
Can you explain why did you remove the `asset_path`?

yihyang replied:
@devonlambert I think you should use asset_path here, otherwise it will not uses the asset pipeline to render the image

wazery commented on lib/generators/ratyrate/templates/ratyrate.js.erb:
```diff
@@ -1,7 +1,39 @@
+  cancel       : false,
+  cancelClass  : 'raty-cancel',
+  cancelHint   : 'Cancel this rating!',
+  cancelOff    : '<%= asset_path('cancel-off.png') %>',
```
Then why do you use asset_path here?

wazery commented:
@devonlambert Great that you splitted the PR into atomic commits, but just a small thing to mention, please change the commit messages to be in an imperative manner, like:

Update ..
Revert ..

To stick to the convention of Git commit messages.

Thanks again for your PR, and your time.

devonlambert commented:
Hi @wazery,

I've added additional information to the commits and updated the subject lines to use the imperative manner. 

In short, the changes I am proposing should make it easier to drop in newer versions of the Raty.js script as it is updated. The way you have it today means that if a new Raty.js version is introduced the "defaults" method would be overridden. Also, Rails was not able to work out the asset_path call in the Raty.js file because you were not using an ERB file.

yihyang commented on lib/generators/ratyrate/ratyrate_generator.rb:
```diff
@@ -7,15 +7,15 @@ class RatyrateGenerator < ActiveRecord::Generators::Base
+    copy_file 'mid-star.png', 'app/assets/images/stars/raty/mid-star.png'
+    copy_file 'big-star.png', 'app/assets/images/stars/raty/big-star.png'
+    copy_file 'cancel-on.png', 'app/assets/images/stars/raty/cancel-on.png'
+    copy_file 'cancel-off.png', 'app/assets/images/stars/raty/cancel-off.png'
```
@devonlambert are you trying to change the location of the icons? If so I think u need to update the location in the rendering files...

devonlambert replied:
@yihyang I am trying to change the location of the icons. However I didn't believe there was a need to change the rendering file as Rails Asset pipeline is smart enough to work out the "asset_path"?

Isn't that the case?
